src/main/java/use_cases/initializer/InitializeUseCase.java

```diff
@@ -0,0 +1,40 @@
+package use_cases.initializer;
+
+import entities.Item;
+import entities.TempDataStorage;
+import use_cases.gateway_interfaces.GatewayReaderInterface;
+
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public class InitializeUseCase implements InitializerInputBoundary {
+
+    public InitializeUseCase(){
+    }
+
+    /**
+     * Initializes TempDataStorage based on the database
+     * @param reader a GatewayReader that reads the database and returns its contents in a List
+     */
+    @Override
+    public void initializeInventory(GatewayReaderInterface reader) throws ParseException, IOException {
+        InitializerDS database = new InitializerDS(reader.getData());
+        Map<String, Item> inventory = new HashMap<>();
+        for (String[] row: database.getInventory()
+             ) {
+            List<String> categories = new ArrayList<>();
+            for (String category: row[4].split(";") //Change depending on how categories are formatted in database
+                 ) {
+                categories.add(category);
+            }
+            DateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
+            Date date = dateFormat.parse(row[5]);
+            Item item = new Item(row[0], row[1], Double.parseDouble(row[2]), Integer.parseInt(row[3]), categories, date, row[6]);
```

Comment:
maybe here we could add an ItemBuilder or ItemFactory to do this job? I realize the constructor for the item class is really long and probably difficult to interpret.